exercises/react/src/example/kotlin/React.kt

```diff
@@ -0,0 +1,72 @@
+class Reactor<T>() {
+    abstract inner class Cell {
+        abstract val value: T
+        internal val dependents = mutableListOf<ComputeCell>()
+        internal fun eachDependent(f: (ComputeCell) -> Any) {
+            for (dep in dependents) {
+                f(dep)
+            }
+        }
+    }
+
+    interface Subscription {
+        fun cancel()
+    }
+
+    inner class InputCell(initialValue: T) : Cell() {
+        override var value: T = initialValue
+            set(newValue) {
+                field = newValue
+                eachDependent { it.propagate() }
+                eachDependent { it.fireCallbacks() }
+            }
+    }
+
+    inner class ComputeCell private constructor(val newValue: () -> T) : Cell() {
+        override var value: T = newValue()
+            private set
+
+        private var lastCallbackValue = value
+        private var callbacksIssued = 0
+        private val activeCallbacks = mutableMapOf<Int, (T) -> Any>()
+
+        constructor(c: Cell, f: (T) -> T) : this({ f(c.value) }) {
+            c.dependents.add(this)
+        }
+        constructor(c1: Cell, c2: Cell, f: (T, T) -> T) : this({ f(c1.value, c2.value) }) {
```

Comment:
> unfortunately I don't believe you can specify a vararg for lambda parameters

Yeah I didn't find one either, nor a way to make sure that the 1-cell form is always with a 1-argument lambda, a 2-cell form with a 2-argument lambda.

Is there a better way to perhaps go about these constructors for increased flexibility? Would passing a collection make more sense?

I believe this is possible. The disadvantage is that we lose a bit of compiler verification: Now someone can pass in two cells with a function that expects to get one value (it only reads the first value out of its collection of arguments). The advantage of course is more flexibility - we can imagine a compute cell that takes 3 cells, etc. even though it is not tested.

What I'm going to do is, if possible, I will make a separate commit that explores using a collection. We can decide whether to have that commit or not have it, when we see what it looks like.